controllers/servers/host_definer/watcher/watcher_helper.py

```diff
@@ -0,0 +1,263 @@
+
+    def get_host_definition_manifest(self, host_definition):
+        return {
+            'apiVersion': settings.CSI_IBM_API_VERSION,
+            'kind': settings.HOST_DEFINITION_KIND,
+            'metadata': {
+                'name': host_definition.name,
+            },
+            'spec': {
+                'hostDefinition': {
+                    settings.MANAGEMENT_ADDRESS_FIELD: host_definition.management_address,
+                    settings.NODE_NAME_FIELD: host_definition.node_name,
+                    settings.NODE_ID_FIELD: host_definition.node_id,
+                    settings.SECRET_NAME_FIELD: host_definition.secret_name,
+                    settings.SECRET_NAMESPACE_FIELD: host_definition.secret_namespace,
+                },
+            },
+        }
+
+    def _get_prefix(self):
+        return os.getenv('PREFIX')
+
+    def get_connectivity(self):
+        return os.getenv('CONNECTIVITY')
+
+    def undefine_host_and_host_definition_with_events(self, host_definition):
+        node_name = host_definition.node_name
+        logger.info('Verifying that host {} is undefined from storage {}'.format(node_name,
+                                                                                 host_definition.management_address))
+        response = self.undefine_host_and_host_definition(host_definition)
+        if response.error_message:
+            self.set_host_definition_status(host_definition.name, settings.PENDING_DELETION_PHASE)
+            self._create_event_to_host_definition(host_definition.name, response.error_message)
+
+    def _create_event_to_host_definition(self, host_definition_name, message):
+        host_definition, _ = self._get_host_definition(host_definition_name)
+        if host_definition:
+            self.add_event_to_host_definition(host_definition, message)
+
+    def get_node_name_from_node_id(self, node_id):
+        node_name, _, _, _ = utils.get_node_id_info(node_id)
+        return node_name
+
+    def add_event_to_host_definition(self, host_definition, message):
+        logger.info('Creating event for host definition: {} error event: {}'.format(host_definition.name, message))
+        event = self._get_event_for_object(host_definition, message)
+        self.create_event(settings.DEFAULT_NAMESPACE, event)
+
+    def is_host_can_be_defined(self, node_name):
+        if self.is_dynamic_node_labeling_allowed():
+            return True
+        return self.is_node_has_managed_by_host_definer_label(node_name)
+
+    def is_dynamic_node_labeling_allowed(self):
+        return os.getenv('DYNAMIC_NODE_LABELING') == 'true'
+
+    def is_host_can_be_undefined(self, node_name):
+        if self._is_host_definer_can_delete_hosts():
+            return self.is_node_has_managed_by_host_definer_label(node_name) and \
+                (not self._is_node_has_host_definer_avoid_deletion_label(node_name))
+        return False
```

Comment:
```suggestion
       return self._is_host_definer_can_delete_hosts() and self.is_node_has_managed_by_host_definer_label(
            node_name) and not self._is_node_has_host_definer_avoid_deletion_label(node_name)
```

Reply:
like there - if `_is_host_definer_can_delete_hosts` is `False` it wont proceed to check the next function..